draft-ietf-tls-ctls.md

```diff
@@ -130,37 +130,69 @@ affected.  We do not show the structures which only change in this way.
 
 ## Record Layer
 
-The cTLS Record Layer assumes that records are externally framed
-(i.e., that the length is already known because it is carried in a UDP
-datagram or the like). Depending on how this was carried, you might
-need another byte or two for that framing. Thus, only the type byte
-need be carried and TLSPlaintext becomes:
+The only cTLS records that are sent in plaintext are handshake records
+(ClientHello and ServerHello/HRR).  The content type is therefore constant (it
+is always `handshake`), so we instead set the `content_type` field to a fixed
+cTLS-specific valud to distinguish cTLS plaintext records from encrypted
+records, TLS/DTLS records, and other protocols using the same 5-tuple.
+
+The `profile_id` field allows the client and server to agree on which
+compression profile should be used for this session (see
+{{template-based-specialization}}).  This field MUST be set to zero if and only
+if no compression profile is used.  Non-zero values are negotiated out of band
+between the client and server, as part of the specification of the compression
+profile.
 
 ~~~~
       struct {
-          ContentType type;
-          opaque fragment[TLSPlaintext.length];
-      } TLSPlaintext;
+          ContentType content_type = ctls;
+          varint profile_id; 
```

Comment:
Do these have to match? What comes from the server?

Reply:
We would only need a profile ID transmitted in the ClientHello. As such, I wonder whether we should omit the profile_id for ServerHellos?

Reply:
Maybe an open issue marker?

Reply:
Done.